packages/button/src/LionButton.js

```diff
@@ -207,13 +211,30 @@ export class LionButton extends DisabledWithTabIndexMixin(SlotMixin(LitElement))
    * Delegate click, by flashing a native button as a direct child
    * of the form, and firing click on this button. This will fire the form submit
    * without side effects caused by the click bubbling back up to lion-button.
-   * @param {Event} e
+   * @param {Event} ev
    */
-  __clickDelegationHandler(e) {
-    if ((this.type === 'submit' || this.type === 'reset') && e.target === this && this._form) {
-      e.stopImmediatePropagation();
-      this._nativeButtonNode.click();
+  __clickDelegationHandler(ev) {
+    if ((this.type === 'submit' || this.type === 'reset') && ev.target === this && this._form) {
+      const nativeButton = document.createElement('button');
+      nativeButton.type = this.type;
+      // Prevents that someone who listens outside of the form catches this event
+      const preventEventLeakage = (/** @type {Event} */ e) => {
+        if (e.target === nativeButton) {
+          e.stopImmediatePropagation();
+        }
+      };
+      this._form.addEventListener('click', preventEventLeakage);
+      this._form.appendChild(nativeButton);
+      // Form submission or reset will happen
+      nativeButton.click();
+      this._form.removeChild(nativeButton);
+      this._form.removeEventListener('click', preventEventLeakage);
     }
+
+    // if ((this.type === 'submit' || this.type === 'reset') && e.target === this && this._form) {
```

Comment:
Remove this?

Reply:
Yes, I already fixed that locally 😎  Could you check again?